src/common/KokkosKernels_Sorting.hpp

```diff
@@ -395,11 +395,11 @@ TeamBitonicSort2(ValueType* values, PermType* perm, Ordinal n, const TeamMember
             if(elem2 < n)
             {
               //both elements in bounds, so compare them and swap if out of order
-              if(comp(values[elem2], values[elem2]))
+              if(comp(values[elem2], values[elem1]))
               {
-                ValueType temp = values[elem1];
+                ValueType temp1 = values[elem1];
```

Comment:
This would have been a good place for a `Kokkos::swap`, as a temporary replacement for `std::swap`.  It could even be that CUDA supports `std::swap` on device now.